ra2ce/analyses/indirect/analyses_indirect.py

```diff
@@ -995,26 +1013,59 @@ def multi_link_isolated_locations(self, graph, analysis, crs=4326):
                 crs=crs,
             )
 
-            # remove the largest (main) graph
+            # remove the largest (main) graph, leaving isolated network
             results.sort_values(by="count", ascending=False, inplace=True)
             results = results.loc[results["count"] != results["count"].max()]
 
             # reproject the datasets to be able to make a buffer in meters
             results = results.set_crs(crs=crs)
             results.to_crs(crs=nearest_utm, inplace=True)
 
+            # process network to get output
+            # get directly disrupted roads
+            results_flooded_roads = gpd.GeoDataFrame(
+                {
+                    "geometry": [e[-1]["geometry"] for e in edges_remove],
+                    "edge_fid": [
+                        f"{e[-1]['node_A']}_{e[-1]['node_B']}" for e in edges_remove
+                    ],
+                },
+                crs=crs,
+            )
+            results_flooded_roads.to_crs(crs=nearest_utm, inplace=True)
+            results_flooded_roads.geometry = results_flooded_roads.geometry.buffer(
+                analysis["buffer_meters"],
+                cap_style=3,
+            )
+            results_flooded_roads[f"i_type_{hazard_name[:-3]}"] = "flooded"
+
+            # get indirect isolated roads
             results_buffered = results.copy()
             results_buffered.geometry = results_buffered.geometry.buffer(
-                analysis["buffer_meters"]
+                analysis["buffer_meters"],
+                cap_style=3,
             )
```

Comment:
why are you 'buffering' again? Please explain why it's required to buffer twice (we have already done this in line 1036)

Reply:
In this analysis, there are two type of networks: flooded and isolated. The buffer is applied accordingly for each type. The code has been improved for readability with regarding to this point.